### PR TITLE
Update CLI Guide with new user flow

### LINKED ISSUE #223: Docs: update the documentation with the new CLI
### What do you want to share with us?

**! This is related to the feature/roles-simplification branch !**

This task is about updating the `docs/source/guide/` for the CLI
As we implement the new 'interface' for user interaction, we need to give the correct screens in the Guide.

We need an update: https://github.com/vmware/repository-service-tuf-cli/blob/feature/roles-simplification/docs/source/guide/index.rst

### References

 - https://github.com/vmware/repository-service-tuf/issues/207

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kairoaraujo**
The CLI Guide is outdated after the new flow for adding the single online key.

This commit updates it.

Basically, it is a new copy from the CLI interface.
In some of `.. code::` the `shell` statement was removed due to incompatibility (special characters in the interface)

Closes #223

### CODE REVIEW

File: docs/source/guide/index.rst [hunk 1]
```diff
@@ -341,9 +369,10 @@ If you choose ``-b/--bootstrap`` it will automatically send the bootstrap to
 
 If you did the ceremony in a disconnected computer:
 Using another computer with access to ``repository-service-tuf-api``
-1.  Get the generated ``payload.json`` (or the custom name you chose)
-2.  Install ``repository-service-tuf``
-3.  Run ``rstuf admin ceremony -b [-u filename]``
+
+  1.  Get the generated ``payload.json`` (or the custom name you chose)
+  2.  Install ``repository-service-tuf``
+  3.  Run ``rstuf admin ceremony -b [-u filename]``
```
Comment (MVrachev) [inline docs/source/guide/index.rst:375]: Isn't the `u` option required to upload it online?
As it seems that you document it as optional or am I wrong?

**Comment by kairoaraujo** [comment]
> I left some suggestions.
> I want to remind you that if we do any changes in the documentation for the ceremony it will be good to also include them in the `ceremony.py` file as well.

What is inside `..code::` blocks are copied/pasted from `rstuf admin ceremony` (`ceremony.py`) CLI.
What you're suggesting is changing the `ceremony.py`, is it correct?

File: docs/source/guide/index.rst [hunk 2]
```diff
@@ -134,204 +133,233 @@ Step 1: Configure the Roles
...
+    That is why the targets role
+    delegates trust to the hash bin roles to reduce the metadata overhead for
+    clients.
 
-    Additionally, it has two sub-folders, productA and productB where the clients can find all files
-    (i.e.: productA-v1.0.tar, productB-v1.0.tar), for productB it even has a sub-folder, updates where
-    clients can find update files (i.e.: servicepack-1.tar, servicepack-2.tar).
-    The organization has decided to use 8 hash bins. Target files will be uniformly distributed over
-    8 bins whose names will be "1.bins-0.json", "1.bins-1.json", ... , "1.bins-7.json".
+    See: TUF Specification about succinct hash bin delegation.
```
[the remark this answers is not in the document]
Comment (kairoaraujo) [reply docs/source/guide/index.rst:166]: This is only a "screenshot". The CLI has a link.

File: docs/source/guide/index.rst [hunk 3]
```diff
@@ -100,23 +102,20 @@ You can do the Ceremony offline. This means on a disconnected computer
...
+    Usage: rstuf admin ceremony [OPTIONS]
+
+    Start a new Metadata Ceremony.
+
+    ╭─ Options ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
+    │ --bootstrap  -b        Bootstrap a Repository Service for TUF using the Repository Metadata after Ceremony                       │
+    │ --file       -f  TEXT  Generate specific JSON Payload compatible with TUF Repository Service bootstrap after Ceremony            │
+    │                        [default: payload.json]                                                                                   │
+    │ --upload     -u        Upload existent payload 'file'. Requires '-b/--bootstrap'. Optional '-f/--file' to use non default file.  │
+    │ --save       -s        Save a copy of the metadata locally. This option saves the metadata files (json) in the 'metadata' folder │
```
[the remark this answers is not in the document]
Comment (kairoaraujo) [reply docs/source/guide/index.rst:115]: Because there is no default (once save is a boolean option).

File: docs/source/guide/index.rst [hunk 4]
```diff
@@ -134,204 +133,233 @@ Step 1: Configure the Roles
...
-    Enter 1/1 the snapshot`s Key path: tests/files/snapshot1.key
-    Enter 1/1 the snapshot`s Key password:
-    ✅ Key 1/1 Verified
+It is essential to define the key owners. There is a suggestion in the CLI.
 
-    Enter 1/1 the timestamp`s Key path: tests/files/timestamp1.key
-    Enter 1/1 the timestamp`s Key password:
-    ✅ Key 1/1 Verified
+The owners will need to be present to insert their keys and use their passwords
+to load their keys.
```
Comment (MVrachev) [inline docs/source/guide/index.rst:239]: I just noticed we mention `their` three times in a row now.
Maybe we can do something like:
```suggestion
The owners will need to be present to insert their keys and passwords
to load their keys.
```

File: docs/source/guide/index.rst [hunk 5]
```diff
@@ -134,204 +133,233 @@ Step 1: Configure the Roles
...
+    The TUF root role supports multiple keys and the threshold (quorum of trust)
+    defines the minimal number of keys required to take actions using the root role.
 
-    Reference: TUF                                                                  
+    Reference: TUF Goals for PKI
 
-    What is the Metadata expiration for the root role?(Days) (365):
+    The TUF roles have an expiration, clients must not trust expired metadata.
+
+    Reference: TUF expires
```
[the remark this answers is not in the document]
Comment (MVrachev) [reply docs/source/guide/index.rst:148]: I am sorry, but I cannot find the `screenshots`.
I searched for them, but they are nowhere to be found.

File: docs/source/guide/index.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (kairoaraujo) [reply docs/source/guide/index.rst:166]: Meaning that we mentioned it already in CLI. What you see in this documentation is only "screenshots" (ASCII copy/paste) from the rendered CLI interface.
https://github.com/vmware/repository-service-tuf-cli/blob/5eb03bf05ab5e345e81f8bef3c324993836164e2/repository_service_tuf/cli/admin/ceremony.py#L200-L203
Screenshot 2023-03-27 at 16 52 10

File: docs/source/guide/index.rst [hunk 6]
```diff
@@ -134,204 +133,233 @@ Step 1: Configure the Roles
...
+    ┃                         STEP 1: Configure the Roles                          ┃
+    ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
 
-    The TUF roles support multiple keys and the threshold (quorum trust) defines
-    the minimal number of keys required to take actions using a specific Role.
+    The TUF root role supports multiple keys and the threshold (quorum of trust)
+    defines the minimal number of keys required to take actions using the root role.
 
-    Reference: TUF                                                                  
+    Reference: TUF Goals for PKI
```
[the remark this answers is not in the document]
Comment (kairoaraujo) [reply docs/source/guide/index.rst:144]: We give the links.
This is just an example from the interface screen the user will see running the CLI. Take it as a "screenshot" for reference.
Screenshot 2023-03-27 at 16 53 47

File: docs/source/guide/index.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (MVrachev) [reply docs/source/guide/index.rst]: Well, it seems there is a default value and the `show_default` option is `True` for `save`:
https://github.com/vmware/repository-service-tuf-cli/blob/5eb03bf05ab5e345e81f8bef3c324993836164e2/repository_service_tuf/cli/admin/ceremony.py#L719

[the remark this answers is not in the document]
Comment (kairoaraujo) [reply docs/source/guide/index.rst]: It is an `is_flag`.
https://github.com/vmware/repository-service-tuf-cli/blob/5eb03bf05ab5e345e81f8bef3c324993836164e2/repository_service_tuf/cli/admin/ceremony.py#L721

An `is_flag` option is a boolean. It means if the user doesn't use `-save` it is False by default.
The user must give `-save` to save the JSON metadata.
It doesn't act like an option `-save true`, `-save false`
The option `-save` acts as a [switch ](https://click.palletsprojects.com/en/8.1.x/options/#feature-switches) if the user wants to save, they must use the flag.

File: docs/source/guide/index.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (MVrachev) [reply docs/source/guide/index.rst:144]: I see the box in which you document those stuff doesn't support rich text format or in other words text with an embedded link.

**Review by MVrachev — CHANGES_REQUESTED** [review]

**Review by MVrachev — APPROVED** [review]

**Pull request merged**